src/DataHandling/Logs/DataDogAudit.php

```diff
@@ -165,22 +271,242 @@ protected function removeColumnIfOnlyUnchanged($diffArray)
 
     /**
      * Method for getting key storing information about given on input column name with association. Can be override for customization needs.
-     * @param \DataDog\AuditBundle\Entity\AuditLog $currentVersion
-     * @return string name 
+     *
+     * @param AuditLog $currentVersion
+     *
+     * @return string name
      */
     protected function getColumnNameForAssociation(AuditLog $currentVersion)
     {
-        return $currentVersion->getTarget()->getTbl();
+        $joinTableName = $currentVersion->getTbl();
+        if (isset($this->cache['assocTable'][$joinTableName])) {
+            return $this->cache['assocTable'][$joinTableName];
+        }
+
+        foreach ($this->getEntitiesClassMetaData()->getAssociationMappings() as $assMapping) {
+            if (isset($assMapping['joinTable'])) {
+                $assJoinTable = $assMapping['joinTable']['name'];
+                $this->cache['assocTable'][$assJoinTable] = $assMapping['fieldName'];
+                if ($joinTableName === $assJoinTable) {
+                    return $assMapping['fieldName'];
+                }
+            } elseif (is_null($currentVersion->getTarget())
+                && $currentVersion->getSource()->getClass() === $assMapping['targetEntity']
+            ) { // ManyToOne mapping
+                $this->cache['assocTable'][$joinTableName] = $assMapping['fieldName'];
+
+                return $assMapping['fieldName'];
+            }
+        }
+
+        throw new \LogicException(sprintf('no association for %s found.', $joinTableName));
```

Comment:
In connection to line 298, it looks really strange that at the end of function we have Exception thrown. Generally, at the end of the function should be return statement (the only one in function body).

Reply:
https://symfony.com/doc/current/contributing/code/standards.html#structure does not tell anything about only one return value.

There are even two return in the example, in `reverseBoolean()`